tests/test_validate_data.py

```diff
@@ -129,26 +136,54 @@ def test_DataValidator_apply_fails(simple_df, file, item_1, item_2, item_3, capl
     assert failed_validation_message in caplog.text
 
 
-def test_DataValidator_validate_with_warning(simple_df, caplog):
+@pytest.mark.parametrize(
+    "file, value",
+    [("joined", 6.0), ("joined", 3.0), ("legacy", 6.0)],
+)
+def test_DataValidator_validate_with_warning(file, value, simple_df, caplog):
+    """Checks that failed validation rows are printed in log."""
+    simple_df = IamDataFrame(simple_df._data.replace(6.0, value))
```

Comment:
You could also do the following to be more specific:
```suggestion
    simple_df._data.iloc[0] = value
```
(where 0 has to be the right number in the series-representation of the data